### DISCUSSION TITLE
Don't require go.mod to be in top directory of workspace.

### DISCUSSION THREAD

**Discussion by jimgrier** [golang/vscode-go#1242] [Discussions]
If the top level directory of a workspace is not where the go.mod file for a package resides, the extension gets confused, flagging imports as errors, etc. Our build layout often has non-Go files of interest in the directories above the Go code, and being able to open at that level would be more convenient.

**Comment by stamblerre** [comment]
@jimgrier: Are you working with multiple Go modules or just a single module? If it is a single module, it does not have to be at the workspace root, but if you are working with multiple modules, there are other challenges. [This documentation](https://github.com/golang/tools/blob/master/gopls/doc/workspace.md) should explain the details and current available options.
